src/PerformanceSensitiveAnalyzers/UnitTests/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.UnitTests.csproj

```diff
@@ -7,6 +7,10 @@
 
     <!-- Do not directly await a task-->
     <NoWarn>$(NoWarn);CA2007</NoWarn>
+
+    <!-- Executable code, so set this in non-source build or repo source-build. Full
+         source build will use the incoming, pre-set MicrosoftCodeAnalysisVersion. -->
+    <MicrosoftCodeAnalysisVersion Condition="'$(DotNetBuildFromSource)' != 'true' or '$(DotNetBuildFromSourceFlavor)' != 'Product'">$(MicrosoftCodeAnalysisVersionForTests)</MicrosoftCodeAnalysisVersion>
```

Comment:
@sharwell I want to verify that in these these unit test cases it is desired to build against the correct surface area (e.g. 2.9.0 for perf sensitive analyzers), but run against the test version (4.6.0*)

Reply:
Looks like these fail when not executed against the 3.3.1 version. My bet is that the failures are just test issues (more diagnostics shown than expected). Reverting for now and marking these two as not building from source with an appropriate comment.